src/linear_algebra/bilinear_form.lean

```diff
@@ -400,3 +410,175 @@ begin
 end
 
 end alt_bilin_form
+
+namespace bilin_form
+
+section linear_adjoints
+
+variables {R : Type u} [comm_ring R]
+variables {M : Type v} [add_comm_group M] [module R M]
+variables {M₂ : Type v} [add_comm_group M₂] [module R M₂]
+variables (B B' : bilin_form R M) (B₂ : bilin_form R M₂)
+variables (f f' : M →ₗ[R] M₂) (g g' : M₂ →ₗ[R] M)
+
+/-- Given a pair of modules equipped with bilinear forms, this is the condition for a pair of
+maps between them to be mutually adjoint. -/
+def is_adjoint_pair := ∀ x y, B₂ (f x) y = B x (g y)
```

Comment:
```suggestion
def is_adjoint_pair := ∀ {{x y}}, B₂ (f x) y = B x (g y)
```
If you use `\{{` in VScode, you get the nice unicode version. These brackets make sure that you don't get annoying metavariables when passing `h` around as an argument. It also enables projection notation.

Reply:
Nice! As always I learn from your review :D